### ISSUE TITLE
gdal_translate works or fails depending on `tr` and `of` parameters

### ISSUE THREAD

**Issue by christophfriedrich** [JoshOBrien/gdalUtilities#15]
A colleague asked me for help with an R script that gathers soil data via `gdal_translate`. This is the minimal reproducible version of it:

```r
library(gdalUtilities)
url <- "/vsicurl?max_retry=3&retry_delay=1&list_dir=no&url=https://files.isric.org/soilgrids/latest/data/ocs/ocs_0-30cm_mean.vrt"
coords <- c(359815.7, 5990580.2, 394170.2, 5960218.9)
crs <- "EPSG:32633"
gdal_translate(url, "test.tif", tr=c(250,250), projwin=coords, projwin_srs=crs)
```

The error received was this:

```
Fehler in gdal_utils("translate", source = src_dataset, destination = dst_dataset,  : 
  gdal_utils translate: an error occured
Zusätzlich: Warnmeldung:
In CPL_gdaltranslate(source, destination, options, oo, config_options,  :
  GDAL Error 11: HTTP response code: 404 - Send failure: Connection was aborted
```

or sometimes this (I can't really tell when we got which version):

```
Fehler in gdal_utils("translate", source = src_dataset, destination = dst_dataset,  : 
  gdal_utils translate: an error occured
Zusätzlich: Warnmeldung:
In CPL_gdaltranslate(source, destination, options, oo, config_options,  :
  GDAL Error 4: `/vsicurl?max_retry=3&retry_delay=1&list_dir=no&url=https://files.isric.org/soilgrids/latest/data/ocs\./ocs_0-30cm_mean/tileSG-005-054/tileSG-005-054_4-2.tif' does not exist in the file system, and is not recognized as a supported dataset name.
```

Fiddling around with the parameters revealed that it works when changing the target resolution `tr` from `250,250` to `1000,1000`. Also, it worked when changing the output format `of` to `VRT`, regardless of the value of `tr`. So:

```r
gdal_translate(url, "test.tif", tr=c(250,250),   projwin=coords, projwin_srs = crs)   # fails
gdal_translate(url, "test.tif", tr=c(1000,1000), projwin=coords, projwin_srs = crs)   # works
gdal_translate(url, "test.vrt", tr=c(250,250),   projwin=coords, projwin_srs=crs, of="VRT")   # also works
```

I don't really know VRT, but I assume that it "works" because it's only a *link* to the actual data, so the data is not actually retrieved during the command, delaying the problematic request for later? Because once I open the resulting file in QGIS, I get the same error messages there.

So I assumed the problem was indeed the resolution and I also assumed that it's not actually the fault of gdalUtilities, but more likely of GDAL itself.
BUT: Running the corresponding command directly in a shell WORKED!

`gdal_translate -tr 250 250 -projwin 359815.7 5990580.2 394170.2 5960218.9 -projwin_srs "EPSG:32633" "/vsicurl?max_retry=3&retry_delay=1&list_dir=no&url=https://files.isric.org/soilgrids/latest/data/ocs/ocs_0-30cm_mean.vrt" test.tif`

**-> Why does it work in the shell but not via gdalUtilities?**

Caveat: The shell command runs a different executable, as it does not access those delivered with `sf`, but a separate GDAL installation (which in this case was even on a different system where I had GDAL available more easily). So my test doesn't prove that it's the fault of gdalUtilities, but anyway it gave me enough confidence to cautiously raise the issue here and ask for your opinion on it.

The command line system reports `GDAL 3.9.2, released 2024/08/13` for `gdal_translate --version`
On the R system, the value of `sf::sf_extSoftVersion()["GDAL"]` is `3.8.2`
My QGIS is version `3.34.0-Prizren` and reports to be using `GDAL-Version 3.7.2`

**Comment by christophfriedrich** [comment]
I was adding the version numbers to my initial post robotically just for completeness, but then after submitting I realised that the differences there might actually be the answer to my question as the only GDAL where it worked was the newest. So I started looking through the release notes and indeed [for the 3.9.2 release they list](https://github.com/OSGeo/gdal/blob/v3.9.2/NEWS.md#raster-drivers):

> VRT driver:
> - fix reading from virtual overviews when SrcRect / DstRect elements are missing

Which refers to this commit: https://github.com/OSGeo/gdal/commit/61eea4ed9ee50b58387130fb1d0bbf0b93c15a68

I have no clue whether this actually affects the creation of the TIFFs here, but the commit message sounded fitting.

**Comment by JoshOBrien** [comment]
Thanks for the follow up (and admirably detailed issue report).  The command line installation of GDAL on my machine is back at version 3.9.1, and I do get failure with a similar error (" HTTP response code: 404"), so I suspect this really is just a GDAL version issue.

I'm not sure why sf bundles the particular versions of GDAL that it does, but I reckon it could be a while before it gets up to 3.9.1. I just installed a bleeding edge version of the package (1.0.19) from [R-universe](https://r-spatial.r-universe.dev/sf), and it comes with GDAL 3.8.4. That [was released](https://github.com/OSGeo/gdal/releases)  in February, and there have been four GDAL releases since then.

In any case, I'm closing this issue as resolved.

Thanks, and best of luck.
